### ISSUE TITLE
Report when a delegation pattern is used to implement interface containing default method

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#8905]
## Expected Behavior of the rule
For below file in Java
```java
// FILE: A.java
interface A {
    public int defaultValue() { return 0 }
    public fun void abstractFun()
}
```

```kt
class AImpl : A {
    override fun abstractFun() { /* doing some work */ }
}

class B(a: AImpl): A by a
```

should be reported by the rule

## Context
See discussion https://youtrack.jetbrains.com/issue/KT-15226 and https://discuss.kotlinlang.org/t/collision-between-kotlin-class-delegation-and-java-8-default-methods/1994 

One issue found in open source otel android sdk https://github.com/open-telemetry/opentelemetry-android/pull/1464
